src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -86,22 +90,37 @@ public function supportsNormalization($data, $format = null)
     /**
      * {@inheritdoc}
      *
+     * In PHP, The $timezone parameter and the current timezone are ignored when the $time parameter either is a UNIX timestamp (e.g. @946684800) or specifies a timezone (e.g. 2010-01-28T15:00:00+02:00).
+     *
+     * This denormalizer assumes that all DateTimeInterface object returned will have the timezone returned by the getTimezone() method.
+     * Default PHP behavior will occur if the getTimezone() method returns null or $defaultContext[self::FORCE_CONTEXT_TIMEZONE] is set to false.
+     *
      * @throws NotNormalizableValueException
      */
     public function denormalize($data, $class, $format = null, array $context = [])
     {
         $dateTimeFormat = $context[self::FORMAT_KEY] ?? null;
+        $dateTimeFormat = $format ?? $dateTimeFormat;
         $timezone = $this->getTimezone($context);
+        $forceContextTimezone = $this->isForceContextTimezone($context);
+
+        if (true !== $forceContextTimezone && null !== $timezone) {
+            @trigger_error(sprintf('The denormalizer function can produce inconsistent timezone results with "%s" set to false.  Set "%s" to NULL for default PHP behaviour.', self::FORCE_CONTEXT_TIMEZONE, self::TIMEZONE_KEY), E_USER_DEPRECATED);
+        }
 
         if ('' === $data || null === $data) {
-            throw new NotNormalizableValueException('The data is either an empty string or null, you should pass a string that can be parsed with the passed format or a valid DateTime string.');
+            throw new NotNormalizableValueException('The data is either an empty string or null, you should pass a string that can be parsed with the given format or a valid DateTime string.');
```

Comment:
You shouldn't change this here since it's not the purpose of this PR.

Reply:
Agreed,